edd_utils/__init__.py

```diff
@@ -137,6 +137,86 @@ def export_study(session, slug, edd_server='edd.jbei.org', verbose=True):
     return df
 
 
+#Get metadata
+def export_metadata(session, slug, edd_server='edd.jbei.org', verbose=False):
+    '''Export Metadata from EDD as a pandas dataframe'''
+
+    try:
+        lookup_response = session.get(f'https://{edd_server}/rest/studies/?slug={slug}')
+
+    except KeyError:
+        if lookup_response.status_code == requests.codes.forbidden:
+            print('Access to EDD not granted\n.')
+            sys.exit()
+        elif lookup_response.status_code == requests.codes.not_found:
+            print('EDD study was not found\n.')
+            sys.exit()
+        elif lookup_response.status_code == requests.codes.server_error:
+            print('Server error\n.')
+            sys.exit()
+        else:
+            print('An error with EDD export has occurred\n.')
+            sys.exit()
+
+    json_response = lookup_response.json()
+    # Catch the error if study slug is not found in edd_server
+    try: 
+        study_id = json_response["results"][0]["pk"]
+    except IndexError:
+        if json_response["results"] == []:
+            print(f'Slug \'{slug}\' not found in {edd_server}.\n')
+            sys.exit()
+    
+    
+    if verbose:
+        print("Study id is ",study_id)
+    # Get the metadata names
+    export_response = session.get(f'https://{edd_server}/rest/metadata_types/')
+    rainer_get=export_response.json()
+    results=rainer_get['results']
+    names=[] #all names of EDD
```

Comment:
You've moved from having a list of 2-tuples to having two lists; again, what you actually want here is a dictionary. Like `metadata_lookup = {item["pk"]: item["type_name"] for item in results}`. Later in the code where you use `names[pknumbers.index(j)]` you would instead do `metadata_lookup.get(j)`.